src/main/java/games/strategy/engine/framework/ProcessRunnerUtil.java

```diff
@@ -29,21 +30,17 @@ public static void populateBasicJavaArgs(final List<String> commands) {
     populateBasicJavaArgs(commands, System.getProperty("java.class.path"));
   }
 
-  public static void populateBasicJavaArgs(final List<String> commands, final long maxMemory) {
+  public static void populateBasicJavaArgs(final List<String> commands, final Optional<String> maxMemory) {
     populateBasicJavaArgs(commands, System.getProperty("java.class.path"), maxMemory);
   }
 
   static void populateBasicJavaArgs(final List<String> commands, final String newClasspath) {
-    // for whatever reason, .maxMemory() returns a value about 12% smaller than the real Xmx value, so we are going to
-    // add 64m to that to
-    // compensate
-    // final long maxMemory = ((long) (Runtime.getRuntime().maxMemory() * 1.15) + 67108864);
-    final long maxMemory = Memory.getMaxMemoryInBytes();
-    System.out.println("Setting memory for new triplea process to: " + (maxMemory / (1024 * 1024)) + "m");
-    populateBasicJavaArgs(commands, newClasspath, maxMemory);
+    populateBasicJavaArgs(commands, newClasspath, ManagementFactory.getRuntimeMXBean().getInputArguments().stream()
+        .filter(s -> s.toLowerCase().startsWith("-xmx")).map(s -> s.substring(4)).findFirst());
   }
 
-  public static void populateBasicJavaArgs(final List<String> commands, final String classpath, final long maxMemory) {
+  public static void populateBasicJavaArgs(final List<String> commands, final String classpath,
```

Comment:
If you make the above changes, then this method can be made private and you hide the `Optional<String>` detail from the API.